app/src/main/java/com/github/se/signify/ui/screens/challenge/NewChallengeScreen.kt

```diff
@@ -75,41 +99,113 @@ fun NewChallengeScreen(navigationActions: NavigationActions) {
+                                                // Firestore
+                                                userViewModel.removeOngoingChallenge(
+                                                    currentUserId, challenge.challengeId)
+                                                userViewModel.removeOngoingChallenge(
+                                                    challenge.player2, challenge.challengeId)
+                                                challengeViewModel.deleteChallenge(
+                                                    challenge.challengeId)
+                                              },
+                                              modifier =
+                                                  Modifier.testTag("OngoingChallengeCard$index"))
+                                        }
+                                      }
+                                }
                           }
                     }
               }
         }
       })
 }
+
+@Composable
+fun OngoingChallengeCard(
+    challenge: Challenge, // Using the existing Challenge data class
+    onDeleteClick: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+  Card(
+      modifier =
+          modifier
+              .fillMaxWidth()
+              .padding(
+                  horizontal = 8.dp, vertical = 4.dp) // Padding for better separation between cards
+              .border(
+                  1.dp,
+                  Color.Gray,
+                  RoundedCornerShape(16.dp)), // Adding border for better visual separation
+      shape = RoundedCornerShape(16.dp), // Rounded corners
+  ) {
+    Row(
+        modifier =
+            Modifier.fillMaxWidth().padding(16.dp), // Padding inside the card for the content
+        verticalAlignment = Alignment.CenterVertically,
+        horizontalArrangement =
+            Arrangement.SpaceBetween // Space between opponent info and delete button
+        ) {
+          Column(
+              verticalArrangement = Arrangement.Center,
+              modifier = Modifier.weight(1f) // Take up all available space
+              ) {
+                // Opponent information
+                Text(
+                    text = "Opponent: ${challenge.player2}",
+                    fontSize = 18.sp,
+                    color = colorResource(R.color.dark_gray),
+                    fontWeight = androidx.compose.ui.text.font.FontWeight.Bold)
+                Text(
+                    text = "Mode: ${challenge.mode}",
+                    fontSize = 14.sp,
+                    color = colorResource(R.color.dark_gray))
+              }
+          IconButton(
+              onClick = onDeleteClick,
+              modifier =
+                  Modifier.padding(start = 8.dp).testTag("DeleteButton${challenge.challengeId}")) {
+                Icon(
+                    imageVector = Icons.Default.Delete,
+                    contentDescription = "Delete Challenge",
+                    tint = Color.Red)
```

Comment:
I think it's better to change the color of this icon to dark_gray to be more consistent with the Figma design.